lib/mayaUsd/ufe/UsdAttributes.cpp

```diff
@@ -114,11 +116,6 @@ static PXR_NS::UsdAttribute _GetAttributeType(const PXR_NS::UsdPrim& prim, const
 
 Ufe::Attribute::Type UsdAttributes::attributeType(const std::string& name)
 {
-    // If we've already created an attribute for this name, just return the type.
-    auto iter = fAttributes.find(name);
-    if (iter != std::end(fAttributes))
-        return iter->second->type();
```

Comment:
Would be useful if an attribute was heavy to create, but since the latest implementation is a thin wrapper on a USD attribute and a NodeDef, we do not need to optimize.

Since we now have an API to remove attributes, this would have forced us to acknowledge that attributes can be removed directly in USD without using this API, causing that cache to become stale.